### ISSUE TITLE
Batch Size won't increase when decreasing the image size

### ISSUE THREAD

**Issue by gungnir33** [tensorflow/models#10804]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [ yes ] I am using the latest TensorFlow Model Garden release and TensorFlow 2.
- [ yes ] I am reporting the issue to the correct repository. (Model Garden official or research directory)
- [ yes ] I checked to make sure that this issue has not been filed already.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/blob/6172f1130d006208e29fa746c1ddb0b4896944af/research/object_detection/model_lib_v2.py

## 2. Describe the bug

I tried to increase the batch size of Mask-RCNN in my computer with RTX 3080Ti. I can only run 1 batch size with image size 1024, and I change the resizer of image to 640, it can train with 2 batch size, but if I keep decrease image size to 320 or 160, it still cannot train with more than 2 batch size. 
I also tried to disable the shuffle buffer and decrease the depth of fasterRCNN, all these won't increase the batch size.
Hope you can help me to increase the batch size with small image size!

## 3. Steps to reproduce

- train the model with `model_lib_v2.py`
- set the batch size to more than 2

## 4. Expected behavior

Increase the batch size while decrease the image size

## 5. Additional context

This is the config file I use:
`model {
  faster_rcnn {
    number_of_stages: 3
    num_classes: 90
    image_resizer {
      fixed_shape_resizer {
        height: 640
        width: 640
      }
    }
    feature_extractor {
      type: 'faster_rcnn_inception_resnet_v2_keras'
    }
    first_stage_anchor_generator {
      grid_anchor_generator {
        scales: [0.25, 0.5, 1.0, 2.0]
        aspect_ratios: [0.5, 1.0, 2.0]
        height_stride: 16
        width_stride: 16
      }
    }
    first_stage_box_predictor_conv_hyperparams {
      op: CONV
      regularizer {
        l2_regularizer {
          weight: 0.0
        }
      }
      initializer {
        truncated_normal_initializer {
          stddev: 0.01
        }
      }
    }
    first_stage_nms_score_threshold: 0.0
    first_stage_nms_iou_threshold: 0.7
    first_stage_max_proposals: 300
    first_stage_localization_loss_weight: 2.0
    first_stage_objectness_loss_weight: 1.0
    initial_crop_size: 17
    maxpool_kernel_size: 1
    maxpool_stride: 1
    second_stage_box_predictor {
      mask_rcnn_box_predictor {
        use_dropout: false
        dropout_keep_probability: 1.0
        fc_hyperparams {
          op: FC
          regularizer {
            l2_regularizer {
              weight: 0.0
            }
          }
          initializer {
            variance_scaling_initializer {
              factor: 1.0
              uniform: true
              mode: FAN_AVG
            }
          }
        }
        mask_height: 33
        mask_width: 33
        mask_prediction_conv_depth: 0
        mask_prediction_num_conv_layers: 4
        conv_hyperparams {
          op: CONV
          regularizer {
            l2_regularizer {
              weight: 0.0
            }
          }
          initializer {
            truncated_normal_initializer {
              stddev: 0.01
            }
          }
        }
        predict_instance_masks: true
      }
    }
    second_stage_post_processing {
      batch_non_max_suppression {
        score_threshold: 0.0
        iou_threshold: 0.6
        max_detections_per_class: 100
        max_total_detections: 100
      }
      score_converter: SOFTMAX
    }
    second_stage_localization_loss_weight: 2.0
    second_stage_classification_loss_weight: 1.0
    second_stage_mask_prediction_loss_weight: 4.0
    resize_masks: false
  }
}

train_config: {
  batch_size: 16
  num_steps: 200000
  optimizer {
    momentum_optimizer: {
      learning_rate: {
        cosine_decay_learning_rate {
          learning_rate_base: 0.008
          total_steps: 200000
          warmup_learning_rate: 0.0
          warmup_steps: 5000
        }
      }
      momentum_optimizer_value: 0.9
    }
    use_moving_average: false
  }
  gradient_clipping_by_norm: 10.0
  fine_tune_checkpoint_version: V2
  fine_tune_checkpoint: "PATH_TO_BE_CONFIGURED/inception_resnet_v2.ckpt-1"
  fine_tune_checkpoint_type: "classification"
  data_augmentation_options {
    random_horizontal_flip {
    }
  }
}

train_input_reader: {
  label_map_path: "PATH_TO_BE_CONFIGURED/label_map.txt"
  tf_record_input_reader {
    input_path: "PATH_TO_BE_CONFIGURED/train2017-?????-of-00256.tfrecord"
  }
  load_instance_masks: true
  mask_type: PNG_MASKS
}

eval_config: {
  metrics_set: "coco_detection_metrics"
  metrics_set: "coco_mask_metrics"
  eval_instance_masks: true
  use_moving_averages: false
  batch_size: 1
  include_metrics_per_category: true
}

eval_input_reader: {
  label_map_path: "PATH_TO_BE_CONFIGURED/label_map.txt"
  shuffle: false
  num_epochs: 1
  tf_record_input_reader {
    input_path: "PATH_TO_BE_CONFIGURED/val2017-?????-of-00032.tfrecord"
  }
  load_instance_masks: true
  mask_type: PNG_MASKS
}
`

## 6. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 22.04.1
- Mobile device name if the issue happens on a mobile device:
- TensorFlow installed from (source or binary):
- TensorFlow version (use command below):2.3.0
- Python version:3.7
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:11.2
- GPU model and memory:RTX3080Ti

**Comment by sushreebarsa** [comment]
@gungnir33 Could you try with the latest TF version 2.10 and let us know the outcome?
Thank you!

**Comment by gungnir33** [comment]
> @gungnir33 Could you try with the latest TF version 2.10 and let us know the outcome? Thank you!

Hi, @sushreebarsa, I've updated to the latest TF version 2.10 and the problem is still the same.
The batch size of 320 and 160 image size still cannot be bigger than 2

**Comment by sushreebarsa** [comment]
@gungnir33 
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thank you!

**Comment by gungnir33** [comment]
Hi, @sushreebarsa , I use colab to reproduce the issue:  https://colab.research.google.com/drive/1rB6yWaSy0IN6kXshsUw6t3QLOIvm5y2S?usp=sharing
I use the model_lib_v2.py and model_main_v2.py from tensorflow official object detection repo and use my own dataset with tfrecord format. I changed the resize and batch size in the pipeline.config, but I cannot increase the batch size to more than 2 while decreasing the image size.
Also, there's another problem: I added code to set the memory tensoflow uses with code `tf.config.experimental.set_memory_growth(each_device, True)`, this means when I changed the image size, the memory usage should decrease, but the tensorflow always uses the whole memory

**Comment by sushreebarsa** [comment]
@gungnir33 Could you please confirm the reproducible [gist](https://colab.research.google.com/gist/sushreebarsa/f92476bdb2593220b307fb58fa4bdf81/mask_rcnn_test.ipynb#scrollTo=4LzAy99Trxsu) here?
Thank you!

**Comment by gungnir33** [comment]
Hi, @sushreebarsa , I try the same code and files as colab and it can train locally on my computer without error. I think it's the preoblem of colab memory, could you please run it locally?

**Comment by sushreebarsa** [comment]
@gungnir33 Could you please share the error log here to analyze the issue locally?
Thank you!
